### PR TITLE
sql: fix non-deterministic priorities in test

### LINKED ISSUE #6267: sql: Timeout in TestTxnUserRestart
https://circleci.com/gh/cockroachdb/cockroach/16732

```
goroutine 10198 [IO wait]:
net.runtime_pollWait(0x7fe83e9f7540, 0x72, 0xc820eb4c00)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8212852c0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8212852c0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc821285260, 0xc820eb4c00, 0x400, 0x400, 0x0, 0x7fe83e9dd050, 0xc820010048)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820026170, 0xc820eb4c00, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc82129ea80, 0x7fe83e9f99c0, 0xc820026170, 0x5, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:460 +0xcc
crypto/tls.(*Conn).readRecord(0xc8206bd200, 0x1b76217, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:562 +0x2d1
crypto/tls.(*Conn).Read(0xc8206bd200, 0xc8207ef000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:939 +0x167
bufio.(*Reader).fill(0xc820d50240)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Read(0xc820d50240, 0xc8201da020, 0x5, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:207 +0x260
io.ReadAtLeast(0x7fe83e9ea070, 0xc820d50240, 0xc8201da020, 0x5, 0x200, 0x5, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:297 +0xe6
io.ReadFull(0x7fe83e9ea070, 0xc820d50240, 0xc8201da020, 0x5, 0x200, 0xc820a9e140, 0x0, 0x0)
	/usr/local/go/src/io/io.go:315 +0x62
github.com/cockroachdb/pq.(*conn).recvMessage(0xc8201da000, 0xc820a9e140, 0x5aa8b2, 0x0, 0x0)
	/go/src/github.com/cockroachdb/pq/conn.go:953 +0x15a
github.com/cockroachdb/pq.(*conn).recv1Buf(0xc8201da000, 0xc820a9e140, 0x52)
	/go/src/github.com/cockroachdb/pq/conn.go:1003 +0x2f
github.com/cockroachdb/pq.(*conn).recv1(0xc8201da000, 0xc8207b16c0, 0xc820a9e140)
	/go/src/github.com/cockroachdb/pq/conn.go:1024 +0x9b
github.com/cockroachdb/pq.(*conn).simpleExec(0xc8201da000, 0xc820416000, 0x4d, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/pq/conn.go:614 +0x2a1
github.com/cockroachdb/pq.(*conn).Exec(0xc8201da000, 0xc820416000, 0x4d, 0x25cc7c8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/pq/conn.go:864 +0x135
database/sql.(*DB).exec(0xc8208b0e70, 0xc820416000, 0x4d, 0x0, 0x0, 0x0, 0x800701, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/database/sql/sql.go:1035 +0x2c2
database/sql.(*DB).Exec(0xc8208b0e70, 0xc820416000, 0x4d, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/database/sql/sql.go:1009 +0xbe
github.com/cockroachdb/cockroach/sql_test.abortTxn(0xc820450000, 0xc8208b0e70, 0x0)
	/go/src/github.com/cockroachdb/cockroach/sql/txn_restart_test.go:391 +0x253
github.com/cockroachdb/cockroach/sql_test.runTestTxn(0xc820450000, 0xc82097ad20, 0x17ee1a0, 0xf, 0xc8205a3d27, 0xc8208b0e70, 0xc820214000, 0xc820a5c428)
	/go/src/github.com/cockroachdb/cockroach/sql/txn_restart_test.go:362 +0x3f0
github.com/cockroachdb/cockroach/sql_test.TestTxnUserRestart.func2(0xc820214000, 0x18fd620)
	/go/src/github.com/cockroachdb/cockroach/sql/txn_restart_test.go:447 +0x5f
github.com/cockroachdb/cockroach/sql_test.exec(0xc820450000, 0xc8208b0e70, 0xc8205a3f18)
	/go/src/github.com/cockroachdb/cockroach/sql/txn_restart_test.go:330 +0x27f
github.com/cockroachdb/cockroach/sql_test.TestTxnUserRestart(0xc820450000)
	/go/src/github.com/cockroachdb/cockroach/sql/txn_restart_test.go:448 +0x88a
testing.tRunner(0xc820450000, 0x22cb348)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892
```

### PR DESCRIPTION
**Pull request by andreimatei**
fixes #6267

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/6324)

### CODE REVIEW

File: sql/txn_restart_test.go [hunk 1]
```diff
@@ -379,13 +381,13 @@ func runTestTxn(t *testing.T, magicVals *filterVals, expectedErr string,
...
 func abortTxn(t *testing.T, sqlDB *sql.DB, key int) {
 	// Execute all statements in one batch so that the server retries it
 	// automatically in case our txn (the pusher) fails to push the victim. Even
 	// though it starts with a HIGH priority, that's no guarantee it'll always
 	// succeed from the first try.
-	// TODO(andrei): make the priority deterministic here once Radu introduces
-	// that syntax and then we won't need all the statements in one batch.
 	deleteCmd := fmt.Sprintf("DELETE FROM t.test WHERE k = %d", key)
 	if _, err := sqlDB.Exec(
 		fmt.Sprintf("BEGIN; SET TRANSACTION PRIORITY HIGH; %s; COMMIT;", deleteCmd)); err != nil {
```
Comment (tschottdorf) [inline sql/txn_restart_test.go]: Do you want to remove this crutch now? You did remove the TODO for it.

File: sql/txn_restart_test.go [hunk 2]
```diff
@@ -379,13 +381,13 @@ func runTestTxn(t *testing.T, magicVals *filterVals, expectedErr string,
 // that had an intent on that key.
 // This cannot be done as an injected error, since we want the pusher to clean
 // up the intents of the pushee.
+// This function needs to be called from tests that set
```
Comment (tschottdorf) [inline sql/txn_restart_test.go]: I don't understand this comment (here and above).

Comment (bdarnell) [inline sql/txn_restart_test.go]: "This function requires that its caller set ..."

**Comment by RaduBerinde** [comment]
LGTM

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[sql/txn_restart_test.go, line 318 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6324#-KGIhSOQHQ6WSGJlpnFC:-KGIhSOQHQ6WSGJlpnFD:1027069803) ([raw file](https://github.com/cockroachdb/cockroach/blob/23a5645b96c895d2b82e3e983e0b1c1998a25741/sql/txn_restart_test.go#L318)):*
Should we check this and panic? (or is the Context not easily accessible from here)

---

*[sql/txn_restart_test.go, line 393 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6324#-KGIVgRojWF7jx9Coaji-r1-393:-KGIhCrxQqHhVEpg9I17:-951116754) ([raw file](https://github.com/cockroachdb/cockroach/blob/23a5645b96c895d2b82e3e983e0b1c1998a25741/sql/txn_restart_test.go#L393)):*
We still need to set the txn priority to high..

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6324#-:-KGIh_IsygNwxNHiOOLl:676834233)*

**Comment by andreimatei** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[sql/txn_restart_test.go, line 318 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6324#-KGIhSOQHQ6WSGJlpnFC:-KGJ_SqCHdXC0JefIJk-:-811002544) ([raw file](https://github.com/cockroachdb/cockroach/blob/23a5645b96c895d2b82e3e983e0b1c1998a25741/sql/txn_restart_test.go#L318)):*
yeah, the Context is not available here.

---

*[sql/txn_restart_test.go, line 384 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6324#-KGIVgRojWF7jx9Coaji-r1-384:-KGJ_Y8svmeSL4Nfq1QB:514706157) ([raw file](https://github.com/cockroachdb/cockroach/blob/23a5645b96c895d2b82e3e983e0b1c1998a25741/sql/txn_restart_test.go#L384)):*
well technically it's not the "caller" that needs to do something; the singleton server needs to be configure appropriately at the beginning of time. I think the comment is all right...

---

*[sql/txn_restart_test.go, line 393 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6324#-KGIVgRojWF7jx9Coaji-r1-393:-KGJ_FFN7lmkrgOdLrtJ:1938643236) ([raw file](https://github.com/cockroachdb/cockroach/blob/23a5645b96c895d2b82e3e983e0b1c1998a25741/sql/txn_restart_test.go#L393)):*
Toby's right that the comment is not needed any more. And since we no longer need it all in one batch, I've rewritten this function in a more go way.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6324)*

**Comment by RaduBerinde** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[sql/txn_restart_test.go, line 393 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6324#-KGIVgRojWF7jx9Coaji-r1-393:-KGJftxKf3ZJNiGFaqgH:676834233) ([raw file](https://github.com/cockroachdb/cockroach/blob/23a5645b96c895d2b82e3e983e0b1c1998a25741/sql/txn_restart_test.go#L393)):*
LGTM

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6324)*

**Pull request merged**